pkg/appmain/appmain.go

```diff
@@ -28,31 +36,40 @@ var (
 	printVersion = flag.Bool("version", false, "Prints the version and exits")
 )
 
-func Run(name string, version string, logic task.TaskFn) {
+func Run(config Config, logic task.TaskFn) {
 	flag.Parse()
 
+	var err error
+
 	if *printVersion {
-		fmt.Fprintln(os.Stdout, version)
+		fmt.Fprintln(os.Stdout, config.Version)
 		os.Exit(ExitSuccess)
 	}
 
-	err := logger.Configure()
-	if err == nil {
-		logger.Info(name, ", v", version)
+	if err = sentry.Initialize(config.SentryConfig); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(ExitFailure)
+	}
+	defer sentry.Close()
+
+	if err = logger.Configure(); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(ExitFailure)
+	}
+	defer logger.Close()
 
-		// Only available on Windows for cleaning up subprocesses
-		job := newJobObject()
-		defer job.Close()
+	logger.Info(config.Name, ", v", config.Version)
 
-		ctx, _ := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
+	// Only available on Windows for cleaning up subprocesses
+	job := newJobObject()
+	defer job.Close()
 
-		taskManager := task.NewTaskManager(ctx)
-		taskManager.GoFn("AppMain", logic)
-		err = taskManager.Wait()
-	}
+	ctx, _ := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
 
+	taskManager := task.NewTaskManager(ctx)
+	taskManager.GoFn("AppMain", logic)
+	err = taskManager.Wait()
 	if err != nil {
-		fmt.Fprintln(os.Stderr, err)
-		os.Exit(ExitFailure)
+		logger.Error(err)
 	}
```

Comment:
Still need the os.Exit(ExitFailure) here so that the error code is correct when the process exits.